File: src/highlight/detector.ts
import type { LanguageId } from "./types";

const EXT_MAP: Record<string, LanguageId> = {
  ".ts":   "typescript",
  ".tsx":  "typescript",
  ".js":   "typescript",  // tree-sitter-typescript WASM handles JavaScript syntax
  ".jsx":  "typescript",  // tree-sitter-typescript WASM handles JavaScript syntax
  ".mjs":  "typescript",  // tree-sitter-typescript WASM handles JavaScript syntax
  ".cjs":  "typescript",  // tree-sitter-typescript WASM handles JavaScript syntax
  ".py":   "python",
  ".rs":   "rust",
  ".go":   "go",
  ".c":    "c",
  ".h":    "c",
  ".cpp":  "c",  // No C++ WASM available; fallback to C grammar
  ".cc":   "c",  // No C++ WASM available; fallback to C grammar
  ".html": "html",
  ".htm":  "html",
  ".css":  "css",
  ".json": "json",
  ".md":   "markdown",
  ".lua":  "lua",
};

export function detectLanguage(filePath: string | null): LanguageId {
  if (!filePath) return "unknown";
  const ext = filePath.match(/\.[^.]+$/)?.[0]?.toLowerCase() ?? "";
  return EXT_MAP[ext] ?? "unknown";
}

File: src/highlight/detector.rs
use std::path::Path;

pub fn detect_language(file_path: Option<&Path>) -> &'static str {
    let Some(path) = file_path else {
        return "Plain Text";
    };
    
    let ext = path.extension().and_then(|e| e.to_str()).unwrap_or("");
    
    match ext {
        "rs" => "Rust",
        "ts" | "tsx" => "TypeScript",
        "js" | "jsx" => "JavaScript",
        "py" => "Python",
        "rb" => "Ruby",
        "go" => "Go",
        "java" => "Java",
        "c" | "h" => "C",
        "cpp" | "cc" | "hpp" => "C++",
        "sh" | "bash" => "Shell",
        "md" => "Markdown",
        "json" => "JSON",
        "toml" => "TOML",
        _ => "Plain Text",
    }
}
